src/main/java/net/dv8tion/jda/internal/requests/restaction/GuildScheduledEventActionImpl.java

```diff
@@ -0,0 +1,201 @@
+    @CheckReturnValue
+    public GuildScheduledEventActionImpl setName(@Nonnull String name)
+    {
+        Checks.notBlank(name, "Name");
+        name = name.trim();
+        Checks.notEmpty(name, "Name");
+        Checks.notLonger(name, GuildScheduledEvent.MAX_NAME_LENGTH, "Name");
+        this.name = name;
+        return this;
+    }
+
+    @Nonnull
+    @Override
+    @CheckReturnValue
+    public GuildScheduledEventActionImpl setDescription(@Nullable String description)
+    {
+        if (description != null)
+            Checks.notLonger(description, GuildScheduledEvent.MAX_DESCRIPTION_LENGTH, "Description");
+        this.description = description;
+        return this;
+    }
+
+    @Nonnull
+    @Override
+    public GuildScheduledEventAction setImage(@Nullable Icon icon)
+    {
+        this.image = icon;
+        return this;
+    }
+
+    @Nonnull
+    @Override
+    public GuildScheduledEventAction setLocation(@Nonnull StageChannel stageChannel)
+    {
+        this.channelId = stageChannel.getIdLong();
+        this.entityType = 1;
+        return this;
+    }
+
+    @Nonnull
+    @Override
+    public GuildScheduledEventAction setLocation(@Nonnull VoiceChannel voiceChannel)
+    {
+        this.channelId = voiceChannel.getIdLong();
+        this.entityType = 2;
+        return this;
+    }
+
+    @Nonnull
+    @Override
+    public GuildScheduledEventAction setLocation(@Nonnull String externalLocation)
+    {
+        this.location = externalLocation;
+        this.entityType = 3;
+        return this;
+    }
+
+    @Nonnull
+    @Override
+    public GuildScheduledEventAction setStartTime(@Nonnull TemporalAccessor startTime)
+    {
+        Checks.notNull(startTime, "Temporal");
+        this.startTime = Helpers.toOffsetDateTime(startTime);
+        return this;
+    }
+
+    @Nonnull
+    @Override
+    public GuildScheduledEventAction setEndTime(@Nullable TemporalAccessor endTime)
+    {
+        Checks.notNull(endTime, "Temporal");
+        this.endTime = Helpers.toOffsetDateTime(endTime);
+        return this;
+    }
+
+    @Override
+    protected RequestBody finalizeData()
+    {
+        preChecks();
+        DataObject object = DataObject.empty();
+        object.put("entity_type", entityType);
+        object.put("privacy_level", 2);
+        object.put("name", name);
+        object.put("scheduled_start_time", startTime.format(DateTimeFormatter.ISO_DATE_TIME));
+        if (entityType == 1 || entityType == 2)
+            object.put("channel_id", channelId);
+        if (entityType == 3 && location != null && location.length() > 0)
+            object.put("entity_metadata", DataObject.empty().put("location", location));
+        if (description != null)
+            object.put("description", description);
+        if (image != null)
+            object.put("image", image.getEncoding());
+        if (endTime != null)
+            object.put("scheduled_end_time", endTime.format(DateTimeFormatter.ISO_DATE_TIME));
```

Comment:
Again missing null checks

Reply:
I added some potentially missing null checks in the preChecks() method. Let me know if I missed any specifically.